components/style/style_adjuster.rs

```diff
@@ -70,6 +72,9 @@ impl<'a, 'b: 'a> StyleAdjuster<'a, 'b> {
         if !flags.contains(CascadeFlags::SKIP_ROOT_AND_ITEM_BASED_DISPLAY_FIXUP) {
             blockify_if!(flags.contains(CascadeFlags::IS_ROOT_ELEMENT));
             blockify_if!(layout_parent_style.get_box().clone_display().is_item_container());
+            if blockify {
+                self.cacheable = false;
```

Comment:
even if `blockify` is true, we could have the same `display`, right?

Reply:
Also, I think there can be the case of a pseudo-element with the `SKIP_ROOT_AND_ITEM_BASED_DISPLAY_FIXUP` flag matching the same rules as a non-pseudo, which would make the optimization non-idempotent I fear...

Reply:
It wouldn't, because even if we use the cached computed values, those values would go through these adjustments again, and thus they would get blockified at that time.